### PR TITLE
Minor documentation updates

### PR DESCRIPTION
**Pull request by cgmb**
There's four main changes.

1. Clean up the rocSOLVER index and introduction text
2. Update rocBLAS URLs
3. Add `rocblas_layer_mode_flags` to the Sphinx docs
4. Update  copyright year in LICENSE.md

### CODE REVIEW

**Review by jzuniga-amd — APPROVED** [review]

File: docs/source/userguide_memory.rst [hunk 1]
```diff
@@ -39,9 +38,8 @@ By default, rocSOLVER will automatically allocate device memory to be used as in
 using the rocBLAS memory model, and will increase the amount of allocated memory as needed by rocSOLVER
 functions. If this scheme is in use, the function ``rocblas_is_managing_device_memory`` will return
 ``true``. In order to re-enable this scheme if it is not in use, a ``nullptr`` or zero size can be
-passed to the helper functions ``rocblas_set_device_memory_size`` or ``rocblas_set_workspace``.
-For more details on these rocBLAS APIs, see the
-`rocBLAS documentation <https://rocblas.readthedocs.io/en/latest/functions.html#device-memory-functions>`_.
+passed to the helper functions ``rocblas_set_device_memory_size`` or ``rocblas_set_workspace``. For
+more details on these rocBLAS APIs, see `Device Memory Allocation Functions in rocBLAS`_.
```
Comment (tfalders) [inline docs/source/userguide_memory.rst:42]: Minor issue, but the capitalization in the link is inconsistent with our other links.

**Review by tfalders — APPROVED** [review]
Looks good! Just a couple minor comments.

File: docs/source/userguide_clients.rst [hunk 2]
```diff
@@ -4,9 +4,9 @@
 Clients
 *********
 
-rocSOLVER has an infrastructure for testing and benchmarking similar to that of
-`rocBLAS <https://rocblas.readthedocs.io/en/latest/clients.html>`_, as well as sample code illustrating
-basic use of the library.
+rocSOLVER has an infrastructure for testing and benchmarking similar to that of the
+`rocBLAS testing and benchmarking <https://rocblas.readthedocs.io/en/latest/Programmers_Guide.html#rocblas-benchmarking-testing>`_,
```
[the remark this answers is not in the document]
Comment (cgmb) [reply docs/source/userguide_clients.rst:8]: I was probably thinking imagining an implicit "infrastructure" on the end there.

File: docs/source/userguide_memory.rst (hunk 1 above)
Comment (cgmb) [reply docs/source/userguide_memory.rst:42]: IDK. It matches the title in rocBLAS..

File: docs/source/userguide_clients.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (cgmb) [reply docs/source/userguide_clients.rst]: Fixed.

File: docs/source/userguide_memory.rst (hunk 1 above)
Comment (cgmb) [reply docs/source/userguide_memory.rst:42]: I'd be happy to revisit this in the future, but just in the interests of time, I'm going to go ahead and merge.

Comment (tfalders) [reply docs/source/userguide_memory.rst:42]: Not a big deal. I can live with it either way.

**Pull request merged**